core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -73,8 +73,8 @@ public static ManifestReader read(ManifestFile manifest, FileIO io, Map<Integer,
    * @return a manifest writer
    */
   public static ManifestWriter write(PartitionSpec spec, OutputFile outputFile) {
-    // always use a v1 writer for appended manifests because sequence number must be inherited
-    return write(1, spec, outputFile, null);
+    // always use a v2 writer to preserve sequence numbers, but use null for sequence number so appends inherit
```

Comment:
Does this mean manifests will be written with the v2 schema (i.e. with sequence numbers) even though `TableMetadata` is v1 and the manifest list is written with v1? And this should work because we do a projection on read and sequence number is optional?

Reply:
This is the constructor used to create an append or rewritten manifest, so this makes the manifests that are passed into `FastAppend`, `MergeAppend`, and `RewriteManifests` support sequence numbers. It is needed for the last case: when rewriting a manifest for a v2 table, we need to preserve sequence numbers.

This works for v1 because v1 tables will ignore the new fields.